shared/test/test_varnish_helpers.rb

```diff
@@ -0,0 +1,130 @@
+require 'minitest/autorun'
+require '../../cookbooks/cdo-varnish/libraries/helpers'
+
+# Unit tests for cdo-varnish/libraries/helpers.rb
+class VarnishHelperTest < Minitest::Test
+
+  def test_paths_to_regex
+    ext_regex = paths_to_regex('*.jpg')
+    assert_equal 'req.url ~ "(\.jpg)(\?.*)?$"', ext_regex
+
+    assert_equal ext_regex, paths_to_regex(['*.jpg'])
+    # extension-wildcard with leading slash allowed
+    assert_equal ext_regex, paths_to_regex('/*.jpg')
+
+    exts_regex = paths_to_regex(%w(*.jpg *.png))
+    assert_equal 'req.url ~ "(\.jpg|\.png)(\?.*)?$"', exts_regex
+
+    path_regex = paths_to_regex(%w(path1/* path2/*))
+    assert_equal 'req.url ~ "^/path1/" || req.url ~ "^/path2/"',  path_regex
+  end
+
+  def test_invalid_path_patterns
+    [
+      # Unsupported extension wildcards
+      '/images/*.jpg',
+      '/a??.jpg',
+      '/a*.jpg',
+      # Wildcard-extension
+      '/*.doc*',
+      # Path with wildcard subexpression
+      '/assets/*/images',
+      # Extension-wildcard containing path delimiter
+      '*/images',
+      # Wildcard at both ends
+      '*images*',
+      # Maximum path length
```

Comment:
Tracking bug is here:   https://www.pivotaltracker.com/story/show/104775628

This will become moot, of course, if we no longer rely on Varnish or at
least simplify the config so that it's not doing language header
simplification in C code.  Presumably the same issue happens in the CDN so
I assume the CDN has a solution for this, how does that work for most uers
who aren't using Varnish?

On Fri, Oct 2, 2015 at 5:43 PM, Phil Bogle <phil@code.org> wrote:

> (Again, not needed right away, I will track this in the bug.)
>
> On Fri, Oct 2, 2015 at 5:39 PM, Phil Bogle <phil@code.org> wrote:
>
>> Is there any way we can incorporate those tests in our build in case we
>> change this?  Do you happen to know how they run them?
>>
>> On Fri, Oct 2, 2015 at 5:38 PM, Phil Bogle <phil@code.org> wrote:
>>
>>> yes, I wouldn't insist that you to fix all of this in the current PR; I
>>> just wanted to invest the time now to look at in detail and understand the
>>> issues while I had a chance to look at it.
>>>
>>> I will create a tracking bug that summarizes all of the issues.
>>>
>>> On Fri, Oct 2, 2015 at 5:30 PM, Will Jordan <notifications@github.com>
>>> wrote:
>>>
>>>> In shared/test/test_varnish_helpers.rb
>>>> <https://github.com/code-dot-org/code-dot-org/pull/4207#discussion_r41080914>
>>>> :
>>>>
>>>> > +
>>>> > +  def test_invalid_path_patterns
>>>> > +    [
>>>> > +      # Unsupported extension wildcards
>>>> > +      '/images/*.jpg',
>>>> > +      '/a??.jpg',
>>>> > +      '/a*.jpg',
>>>> > +      # Wildcard-extension
>>>> > +      '/*.doc*',
>>>> > +      # Path with wildcard subexpression
>>>> > +      '/assets/*/images',
>>>> > +      # Extension-wildcard containing path delimiter
>>>> > +      '*/images',
>>>> > +      # Wildcard at both ends
>>>> > +      '*images*',
>>>> > +      # Maximum path length
>>>>
>>>> added in 1a507d2
>>>> <https://github.com/code-dot-org/code-dot-org/commit/1a507d21da4bf1a2881c8ba33e45fc7a96fbabe8>
>>>> .
>>>>
>>>> —
>>>> Reply to this email directly or view it on GitHub
>>>> <https://github.com/code-dot-org/code-dot-org/pull/4207/files#r41080914>
>>>> .
>>>>
>>>
>>>
>>
>